### PR TITLE
WFCORE-6650 Expose server stability and allow feature filtering from a DeploymentUnit.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFCORE-6650

### CODE REVIEW

File: server/src/main/java/org/jboss/as/server/deployment/RootDeploymentUnitService.java [hunk 1]
```diff
@@ -51,7 +53,7 @@ public RootDeploymentUnitService(final Consumer<DeploymentUnit> deploymentUnitCo
                                      final Supplier<DeploymentMountProvider> serverDeploymentRepositorySupplier,
                                      final Supplier<PathManager> pathManagerSupplier,
                                      final Supplier<VirtualFile> contentsSupplier,
-                                     final String name, final String managementName, final DeploymentUnit parent,
+                                     final String name, final String managementName, final DeploymentUnit parent, Stability stability,
```
Comment (yersan) [inline server/src/main/java/org/jboss/as/server/deployment/RootDeploymentUnitService.java:56]: I would suggest adding `Final` or removing all the `Final` modifiers.

In general, it is better if we follow the same pattern, that will avoid any doubts or questions about why ones are final and not others.

**Review by yersan — COMMENTED** [review]
@pferraro This PR will require coordination with WildFly full, link the PR to this one when it is done.

File: server/src/main/java/org/jboss/as/server/deployment/RootDeploymentUnitService.java (hunk 1 above)
Comment (pferraro) [reply server/src/main/java/org/jboss/as/server/deployment/RootDeploymentUnitService.java]: Updated

**Comment by pferraro** [comment]
> @pferraro This PR will require coordination with WildFly full, link the PR to this one when it is done.

I've added a temporary default method implementation to resolve the compilation issue on WF full.

**Review by yersan — APPROVED** [review]
LGTM, @bstansberry Do you have any feedback about this?

@pferraro you could squash related commits.

**Comment by PrarthonaPaul** [comment]
Hello everyone, 
I just wanted to add a note here that I have 2 community-level RFE candidates that use deployment config (oidc.json). So, those PRs are dependent on this PR. I am currently working on converting them into community stability level. 
Here are the WFLY issues for the PRs: 
- OIDC scope RFE: https://issues.redhat.com/browse/WFLY-16532
- OIDC request/request_URI: https://issues.redhat.com/browse/WFLY-17143
Thanks!

**Review by bstansberry — APPROVED** [review]

**Comment by yersan** [comment]
The reposted failure on "Preview Integration Linux - JDK 17" is due to a flaky test, this is ready-to-go

@PrarthonaPaul I plan to release WildFly Core tomorrow with these changes in.

**Comment by yersan** [comment]
Thanks @pferraro / @bstansberry

**Pull request merged**